### PR TITLE
adding javascript reconnect docs

### PR DESCRIPTION
**Pull request by bradygaster**
adding some work to address #6757, though this PR fails to contribute a holistic document addressing the entire topic of reconnections.

### CODE REVIEW

File: aspnetcore/signalr/javascript-client.md [hunk 1]
```diff
@@ -93,6 +93,18 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 [!code-javascript[Logging levels](javascript-client/sample/wwwroot/js/chat.js?range=9-12)]
 
+## Reconnecting Clients
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: As a rule thumb, avoid gerunds in titles and use sentence casing. 

Reconnecting Clients --> Reconnect clients

File: aspnetcore/signalr/javascript-client.md [hunk 2]
```diff
@@ -93,6 +93,18 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 [!code-javascript[Logging levels](javascript-client/sample/wwwroot/js/chat.js?range=9-12)]
 
+## Reconnecting Clients 
+
+The JavaScript client for SignalR doesn't automatically reconnect. As a result, you must write code that will reconnect your client manually. The code below demonstrates a typical reconnection approach:
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: The code below --> The following code

File: aspnetcore/signalr/javascript-client.md [hunk 3]
```diff
@@ -93,6 +93,18 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
...
+
+1. A method (in this case `start` is created to start the connection.
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: Add a closing paren

File: aspnetcore/signalr/javascript-client.md [hunk 4]
```diff
@@ -93,6 +93,18 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
...
+1. When the connection fails (the `catch` handler), wait for a period of time and re-run the `start` method.
+1. Run the `start` method in the connection's `onclose` event handler. 
+
+[!code-javascript[Reconnect the JavaScript client](javascript-client/sample/wwwroot/js/chat.js?range=30-42)]
+
+In a real-world implementation this implementation might be improved upon by changing the code to use an exponential back-off or to retry a specified number of times before giving up.
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: Add a comma after "real-world implementation".

File: aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js [hunk 5]
```diff
@@ -25,4 +25,19 @@ document.getElementById("sendButton").addEventListener("click", event => {
     event.preventDefault();
 });
 
-connection.start().catch(err => console.error(err.toString()));
\ No newline at end of file
+connection.start().catch(err => console.error(err.toString()));
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js]: I noticed there's a *chat.ts* file in this sample app. Thoughts on modifying that instead and letting TS generate this JS file?

File: aspnetcore/signalr/javascript-client.md (hunk 1 above)
Comment (bradygaster) [reply aspnetcore/signalr/javascript-client.md]: i had to google [gerund](http://www.chompchomp.com/terms/gerund.htm), but that's a good suggestion!

File: aspnetcore/signalr/javascript-client.md (hunk 2 above)
Comment (bradygaster) [reply aspnetcore/signalr/javascript-client.md]: got it!

File: aspnetcore/signalr/javascript-client.md (hunk 1 above)
Comment (scottaddie) [reply aspnetcore/signalr/javascript-client.md]: I had to do the same the first time I heard the word. :-)

File: aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js (hunk 5 above)
Comment (bradygaster) [reply aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js]: i have no problem with that but....

> blush

i don't know how to do that..

File: aspnetcore/signalr/javascript-client.md [hunk 6]
```diff
@@ -93,6 +93,18 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 [!code-javascript[Logging levels](javascript-client/sample/wwwroot/js/chat.js?range=9-12)]
 
+## Reconnect Clients 
+
+The JavaScript client for SignalR doesn't automatically reconnect. As a result, you must write code that will reconnect your client manually. The following code demonstrates a typical reconnection approach:
+
+1. A function (in this case `start` function) is created to start the connection.
+1. When the connection fails (the `catch` handler), wait for a period of time and re-run the `start` method.
```
[the remark this answers is not in the document]
Comment (anurse) [reply aspnetcore/signalr/javascript-client.md]: This is a correct interpretation. It's the `onclose` event that fires when the connection fails mid-stream.

[the remark this answers is not in the document]
Comment (bradygaster) [reply aspnetcore/signalr/javascript-client.md]: To minimize confusion, along with the other suggestion @BrennanConroy made, i've changed this bullet list. I've basically removed the second bullet. The point here is, whenever it closes, you re-run the start method to reconnect it. Simplifying the steps made it easier to grok. Let me know if we **need** the mention of the erroring, but it seems like onclose is enough.

File: aspnetcore/signalr/javascript-client.md [hunk 7]
```diff
@@ -93,6 +93,18 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
...
+1. Run the `start` method in the connection's `onclose` event handler. 
+
+[!code-javascript[Reconnect the JavaScript client](javascript-client/sample/wwwroot/js/chat.js?range=30-42)]
+
+In a real-world implementation, this code might be improved upon by changing the code to use an exponential back-off or to retry a specified number of times before giving up.
```
[the remark this answers is not in the document]
Comment (bradygaster) [reply aspnetcore/signalr/javascript-client.md]: Thanks @tdykstra - i knew you'd ping me for `passive voice` usage. I need to get back on the wagon! :)

File: aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js [hunk 8]
```diff
@@ -25,4 +25,17 @@ document.getElementById("sendButton").addEventListener("click", event => {
...
+    } catch (err) {
+        console.log(err);
+        setTimeout(() => start(), 5000);
+    }
+};
+
+connection.onclose(function () {
+    start();
+});
+
```
Comment (BrennanConroy) [inline aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js]: you forgot to call `await start();`

Comment (bradygaster) [reply aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js]: #facepalm - fixed. thanks!

File: aspnetcore/signalr/javascript-client.md [hunk 9]
```diff
@@ -93,6 +93,17 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 [!code-javascript[Logging levels](javascript-client/sample/wwwroot/js/chat.js?range=9-12)]
 
+## Reconnect Clients
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: Clients --> clients

File: aspnetcore/signalr/javascript-client.md [hunk 10]
```diff
@@ -93,6 +93,17 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 [!code-javascript[Logging levels](javascript-client/sample/wwwroot/js/chat.js?range=9-12)]
 
+## Reconnect Clients 
+
+The JavaScript client for SignalR doesn't automatically reconnect. As a result, you must write code that will reconnect your client manually. The following code demonstrates a typical reconnection approach:
+
+1. A function (in this case `start` function) is created to start the connection.
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: Add ", the" after "in this case".

File: aspnetcore/signalr/javascript-client.md [hunk 11]
```diff
@@ -93,6 +93,17 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
...
+1. Run the `start` method in the connection's `onclose` event handler. 
+
+[!code-javascript[Reconnect the JavaScript client](javascript-client/sample/wwwroot/js/chat.js?range=30-42)]
+
+A real-world implementation would use an exponential back-off or to retry a specified number of times before giving up.
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: to retry --> retry

File: aspnetcore/signalr/javascript-client.md [hunk 12]
```diff
@@ -93,6 +93,17 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 [!code-javascript[Logging levels](javascript-client/sample/wwwroot/js/chat.js?range=9-12)]
 
+## Reconnect Clients 
+
+The JavaScript client for SignalR doesn't automatically reconnect. As a result, you must write code that will reconnect your client manually. The following code demonstrates a typical reconnection approach:
+
+1. A function (in this case `start` function) is created to start the connection.
+1. Run the `start` method in the connection's `onclose` event handler.
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: method --> function

File: aspnetcore/signalr/javascript-client.md [hunk 13]
```diff
@@ -93,6 +93,17 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 [!code-javascript[Logging levels](javascript-client/sample/wwwroot/js/chat.js?range=9-12)]
 
+## Reconnect Clients 
+
+The JavaScript client for SignalR doesn't automatically reconnect. As a result, you must write code that will reconnect your client manually. The following code demonstrates a typical reconnection approach:
```
Comment (scottaddie) [inline aspnetcore/signalr/javascript-client.md]: ~~As a result, you must write code that will reconnect your client manually.~~
You must write code that manually reconnects your client.

**Comment by scottaddie** [comment]
@bradygaster Please also update the `ms.date` value to 11/01/2018. This PR is significant enough to justify the update.

File: aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js [hunk 14]
```diff
@@ -25,4 +25,17 @@ document.getElementById("sendButton").addEventListener("click", event => {
     event.preventDefault();
 });
 
-connection.start().catch(err => console.error(err.toString()));
\ No newline at end of file
+const start = async () => {
```
Comment (anurse) [inline aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js]:
```suggestion
async function start() {
```

Comment (bradygaster) [reply aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js]: just curious why to switch this to the "function" syntax when the others are using the () => syntax? i'm okay with it but i'm more okay with consistency.

Comment (anurse) [reply aspnetcore/signalr/javascript-client/sample/wwwroot/js/chat.js]: This is a top-level function, rather than a lambda

**Comment by scottaddie** [comment]
Thank you for the PR, @bradygaster!

**Pull request merged**